web-components/src/components/favorite/Favorite.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Cisco Systems, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import reset from "@/wc_scss/reset.scss";
+import { customElement, html, internalProperty, LitElement, property } from "lit-element";
+import { FocusMixin } from "@/mixins";
+import "@/components/icon/Icon";
+import styles from "./scss/module.scss";
+import { Key } from "@/constants";
+import { nothing } from "lit-html";
+import { classMap } from "lit-html/directives/class-map";
+import { ifDefined } from "lit-html/directives/if-defined";
+
+export namespace Favorite {
+  @customElement("md-favorite")
+  export class ELEMENT extends FocusMixin(LitElement) {
+    @property({ type: Boolean }) disabled = false;
+    @property({ type: Boolean }) checked = false;
+    @property({ type: String }) value = "Select favorite";
+    @property({ type: String }) id = "";
+    @property({ type: String }) label = "Favorite"
+
+    @internalProperty() private customId = "";
+
+    connectedCallback() {
+      super.connectedCallback();
+      this.addEventListener("keydown", this.handleElectKeyDown);
+      this.addEventListener("click", this.handleFavorite as EventListener);
+    }
+
+    static get styles() {
+      return [reset, styles];
+    }
+
+    getIconName() {
+      switch (this.checked) {
+        case true:
+          return "favorite-active_16";
+        default:
+          return "favorite_16";
+      }
+    }
+
+    handleFavorite(event: CustomEvent) {
+      if (this.disabled) {
+        return;
+      } else {
+        this.checked = !this.checked;
+        this.dispatchEvent(
+          new CustomEvent<{value: string, active: boolean}>("favorite-toggle", {
+          detail: {
+              active: this.checked,
+              value: this.value
+          },
+          bubbles: true,
+          composed: true
+          })
+        )
+      }
+    }
+
+    handleElectKeyDown(event: KeyboardEvent) {
+      if (event.code === Key.Enter || event.code === Key.Space) { 
+        this.checked = !this.checked;
+        this.dispatchEvent(
+          new CustomEvent<{value: string, active: boolean}>("favorite-keydown", {
+          detail: {
+              active: this.checked,
+              value: this.value
+          },
+          bubbles: true,
+          composed: true
+          })
+        )
```

Comment:
This looks redundant and effectively emits inconsistent custom events based on how the favorite is toggled. Maybe better to have `handleElectKeydown` trigger the `handleFavorite` function rather than do all the same work with a different customEvent

Reply:
Agreed! Changed.